shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -898,6 +906,27 @@ void onUserLeaveHint() {
     }
   }
 
+  /**
+   * Invoke this from {@code Activity#onWindowFocusChanged()}.
+   *
+   * <p>A {@code Fragment} host must have its containing {@code Activity} forward this call so that
+   * the {@code Fragment} can then invoke this method.
+   */
+  void onWindowFocusChanged(boolean hasFocus) {
+    ensureAlive();
+    if (host.shouldDispatchAppLifecycleState() && flutterEngine != null) {
+      // This may notify in a redundant fashion with onResume and onPause. The
```

Comment:
what's the sequence of the enum's traversal now with this change and without in the most basic case?

Reply:
The sequence is the same, there just might be redundant notifications when the app loses focus and then is made inactive, or is resumed and given focus.